.github/workflows/sample-tests.yaml

```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: sample tests
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  pull_request_target:
+    types: [labeled]
+  schedule:
+  - cron:  '0 2 * * *'
+jobs:
+  # job to run change detection
+  changes:
```

Comment:
Won't these jobs run in parallel? Does the filter step still apply correctly to the java-sample step?

Reply:
Unclear if this is solved -- It looks like the changes runs it it's own check. Should this be inlined for each language?

Reply:
The changes check runs before all the other language specific ones and determines which files were changed. the `needs: changes` in the java check makes sure that job runs after changes and uses its result.